### PR TITLE
Restore - Bugfixes

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: Client/DropboxSdkClient.php [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
+    private $localFilesystem;
+
+    /**
+     *
+     * @param array $params
+     * @param $restoreFolder
+     * @param Filesystem $localFilesystem
      */
-    public function __construct($params)
+    public function __construct(array $params, $restoreFolder, Filesystem $localFilesystem)
```
Comment (jongotlin) [inline Client/DropboxSdkClient.php]: I also like array type hinting :-) https://github.com/dizda/CloudBackupBundle/pull/110#discussion_r84151705

Comment (Nyholm) [reply Client/DropboxSdkClient.php]: haha =) 

Good thing you reviewed it!

Comment (jongotlin) [reply Client/DropboxSdkClient.php]: Wont the other arguments also break BC?

Comment (Nyholm) [reply Client/DropboxSdkClient.php]: True, I have to make them optional. Same thing should be done for other clients. I think I copied these from Gaufrette

**Comment by Nyholm** [comment]
Would you mind fixing the BC breaks in our code?

**Comment by jongotlin** [comment]
Sure!

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**